### PR TITLE
Select2 removal

### PR DESCRIPTION
**Pull request by ymeynot45**
**This pull request is a work in progress**
I wanted to keep you all updated as to how the jQuery removal was going. 
I have chosen [TomSelect](https://tom-select.js.org/) to replace select2.
The difficulty has come where TomSelect uses divs and select2 uses li.  So the css has required a lot of manhandling.
I am looking for thoughts from anyone who is interested in this project.  I will continue to work through it.  I hope this update is useful for anyone else working on the site.

**Currently known errors:**
- there is a bug in TomSelect that I am trying to resolve:
    - The first item searched for will not prioritize the searching by tag name, thus leading to possible incorrect selections if you don't type enough letters to refine the search.

### CODE REVIEW

**Comment by ymeynot45** [comment]
I'm in a holding pattern while I wait for the TomSelect creator to resolve a small bug I found.  He has resolved multiple issues in the last week, so I hope for a quick turn around.

**Review by pushcx — CHANGES_REQUESTED** [review]
Some notes on the work in progress. Mostly this is me being particular about comments, of all the nitpicky things.

File: app/assets/stylesheets/application.css.erb [hunk 1]
```diff
@@ -1434,51 +1433,108 @@ div.flash-success h2 {
...
+  padding: 2px;
+  left: 0px;
+  width: 12px;
+  height: 13px;
+  outline: none;
+  cursor: pointer;
+  color: grey;
+  font-weight: bold;
+  line-height: 13px;
+  margin-left: 0;
```
[the remark this answers is not in the document]
Comment (ymeynot45) [reply app/assets/stylesheets/application.css.erb:1485]: I was able to move half of them and delete the doubles, but some of them break the location of the button if you try to move them to ::after.

File: app/assets/stylesheets/application.css.erb [hunk 2]
```diff
@@ -1434,51 +1433,108 @@ div.flash-success h2 {
...
-.select2-container .select2-results li em {
-	background: none;
-	color: var(--color-fg-contrast-4-5);
-	font-style: italic;
-	padding-left: 0.5em;
-}
-
-.select2-container .select2-results .select2-highlighted em {
-	color: var(--color-box-bg);
+  border: 1px solid var(--color-box-border);
```
[the remark this answers is not in the document]
Comment (ymeynot45) [reply app/assets/stylesheets/application.css.erb:1440]: Sigh* the eternal battle continues.  The line above was a tab and the line below was 2 spaces, so this was no-mans land?
Did you want me to make them all tabs or all 2 spances?

File: app/assets/javascripts/application.js.erb [hunk 3]
```diff
@@ -565,22 +550,18 @@ onPageLoad(() => {
...
-
-      // can't remove the jquery until I replace select 2
-      if ($("#story_url").val().match(new RegExp(u, "i"))) {
-        var ta = $("#story_tags_a").data("select2");
-        if (ta.getVal().indexOf(tag) < 0)
-          ta.addSelectedChoice({ id: tag });
+    const storyUrlEl = document.getElementById('story_url');
+    for (const [match, tag] of Object.entries(url_tags)) {
+      if (storyUrlEl.value.match(new RegExp(match, "i"))) {
+        Lobster.tom.addItem(tag.toLowerCase());
```
[the remark this answers is not in the document]
Comment (ymeynot45) [reply app/assets/javascripts/application.js.erb:554]: Any time you add a tag from the list of possible tags, TomSelect removes it from the list.
Then when TomSelect can't find the tag in the list it won't add it. 
If you add a tag and then delete it, TomSelect puts it back into the list.

**Comment by pushcx** [comment]
Thank you for addressing everything and carrying this project through. I'm going to merge and deploy now while traffic is low.

**Pull request merged**